modules/internal/CPtr.chpl

```diff
@@ -420,17 +420,23 @@ module CPtr {
 
   /*
 
-    Returns a :type:`c_ptr` to a Chapel rectangular array.  Note that the
-    existence of this :type:`c_ptr` has no impact on the lifetime of the array.
-    The returned pointer will be invalid if the original array is freed or even
-    reallocated. Domain assignment could make this :type:`c_ptr` invalid. If
-    the array's data is stored in more than one chunk the procedure will halt
-    the program with an error message.
-
-    :arg arr: the array for which we should retrieve a pointer
-    :returns: a pointer to the array data
+    Returns a :type:`c_ptr` to the elements of a non-distributed
+    Chapel rectangular array.  Note that the existence of this
+    :type:`c_ptr` has no impact on the lifetime of the array.  The
+    returned pointer will be invalid if the original array is freed or
+    even reallocated. Domain assignment could make this :type:`c_ptr`
+    invalid.
+
+    :arg arr: the array for which a pointer should be returned
+    :returns: a pointer to the array's elements
```

Comment:
Does it make sense to say "to the array's first element" here? Or is it too obvious?

Reply:
I don't know if it's "too obvious", but that is an interesting question.  If we had an array that stored its elements in reverse order in memory, I'd expect this to give a pointer to the last element (though I realize the current implementation would fail to do that and still point to the first, which would be the end of the buffer).  So I guess I'd say this phrasing is closer to what the function is intended to do even though it's not precisely what it does at present.

Reply:
I see..

More importantly, though, I think what you are saying is a good argument against being able to pass Chapel arrays to C functions. I have no idea how you can pass an array with such a layout to a C function and expect it to behave well.

Reply:
Well, it depends on what it's doing.  If it's dumping the buffer out to a file, and then later it will read it from the file and pass it back into the same type of Chapel array, it's fine.  But yeah, in general, having arbitrary memory layouts in Chapel makes it challenging to tell a language like C where everything lives with a single pointer.